mempool/src/core_mempool/transaction.rs

```diff
@@ -55,8 +55,8 @@ impl MempoolTransaction {
         self.txn.gas_unit_price()
     }
 
-    pub(crate) fn get_committed_hash(&self) -> HashValue {
-        self.txn.clone().committed_hash()
+    pub(crate) fn get_lookup_hash(&self) -> HashValue {
```

Comment:
It's fine to lose information here? (Why not use the real hash?) 
Is there concern around duplicated user txns from different batches?